low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -190,12 +190,21 @@ def __init__(self, args):
         super().__init__(args)
 
     def validate(self):
-        # Common validator classes to check Cortx/system wide validator
-        pass
+        if self.args and len(self.args) > 1:
+            i=0
+            while i < len(self.args):
+                try:
+                    if self.args[i] == "sanity":
+                        break
+                    elif self.args[i] == "alerts":
+                        break                   
+                except (IndexError, ValueError):
+                    raise SetupError(errno.EINVAL, "Invalid plan for %s"% self.name)
+                i+=1
```

Comment:
Please add validation on incorrect plan.

Also, this can be simplified still.
`if len(self.args) !=2:
    raise SetupError(1, "Validation failure - not enough arguments passed. Please check usage.")

if self.args[1] not in ["sanity", "alerts"]:
     raise SetupError(1, "Invalid plan type specified. %s", self.args[1])`

Reply:
Done the changes.